### ISSUE TITLE
Garbage characters get inserted and eventually pwsh crashes on Linux VM

### ISSUE THREAD

**Issue by jianyunt** [PowerShell/PowerShell#6520]
On my Linux VM, start pwsh, remove psreadline module, and press left or right arrow key and hold, you will see below screen shot.  

![image](https://user-images.githubusercontent.com/10171679/37999446-b8e9d878-31d7-11e8-82e3-de1ab83c6503.png)

pwsh does not seem to interprets left, right arrow keys well (^[OD or ^[[D ^[[C or ^[OC)  . Instead, it inserts OC, OD into the input text.  It kept looping through the code from [line 1593](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs#L1593) to the [line 1808](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs#L1808). Eventually receives escape key, sets ['s' to empty](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs#L1776). Then in the next round of do-while loop, crashes at [line 1798](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs#L1798).

We are encountering the very same behavior in the pwsh bash cloudshell. On the bash, I do not remove psreadline module. Instead, we call register-eventobject or other start-job commands that may make OS busy. After that, OC, OD will be inserted while press and hold L-, R- arrow keys.  

In the above two cases, symptom is the same. They may caused by the same issue in pwsh.

Steps to reproduce
------------------

See above.

Expected behavior
-----------------

1) Key strokes should be deciphered properly. 
2) No crashes

Actual behavior
---------------
1) garbage gets inserted into input text.
2) the pwsh process crashes.

Environment data
----------------

![image](https://user-images.githubusercontent.com/10171679/38000444-74add2a4-31dc-11e8-95c0-f85cfe774fc3.png)

### env $TERM
xterm-256color

### uname -a:
4.4.0-116-generic #140-UBuntu
UBuntu  16.04.1 LTS

**Comment by joeyaiello** [comment]
Hey @jianyunt, is there a reason why you're needing to remove PSReadline. AFAIK, PSReadline is our de facto Readline on non-Windows, there's no supported fallback.

**Comment by jianyunt** [comment]
I am seeing the same behavior on pwsh-bash clouldshell as on Linux without psreadline. That's the reason.

**Comment by JamesWTruher** [comment]
what is the value of $env:TERM?

**Comment by SteveL-MSFT** [comment]
I was able to repro this, but it looks like an issue in coreclr

```none
#0  0x00007ffff6d80428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff6d8202a in __GI_abort () at abort.c:89
#2  0x00007ffff659e1fc in PROCAbort () from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#3  0x00007ffff659d0eb in PROCEndProcess(void*, unsigned int, int) ()
   from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#4  0x00007ffff62420d3 in SafeExitProcess(unsigned int, int, ShutdownCompleteAction) ()
   from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#5  0x00007ffff62436f1 in EEPolicy::HandleFatalError(unsigned int, unsigned long, char16_t const*, _EXCEPTION_POINTERS*) () from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#6  0x00007ffff651fa06 in SystemNative::GenericFailFast(StringObject*, ExceptionObject*, unsigned long, unsigned int) () from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#7  0x00007ffff652019e in SystemNative::FailFastWithException(StringObject*, ExceptionObject*) ()
   from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#8  0x00007fff7f08670b in ?? ()
#9  0x00007fff7f086700 in ?? ()
#10 0x00007fff7d1f7df6 in ?? ()
#11 0x00007fffffffc170 in ?? ()
#12 0x00007fff5855c930 in ?? ()
#13 0x00007fffffff90a0 in ?? ()
#14 0x00007ffff630090d in ExceptionTracker::CallCatchHandler(_CONTEXT*, bool*) ()
   from /opt/microsoft/powershell/6.1.0~preview.1/libcoreclr.so
#15 0x00007fff7d1504c8 in ?? ()
#16 0x0000000000000000 in ?? ()
```

Need some symbols to see all the stack frames

**Comment by jianyunt** [comment]
Potentially this problem might be caused by the screen buffer size in the Linux container for the bash cloudshell case. Still investigation.

**Comment by raghushantha** [comment]
Key bindings work as expected when xterm is set to "xterm-256color" (xterm 3.2.0) in https://rc.portal.azure.com

What are the other issues when using xterm-256color?

**Comment by jianyunt** [comment]
By pressing and holding  left or right arrow keys you will see ascii key code (OC, OD) gets inserted into your input string. It happens on xterm-color or xterm-256color and the latest bits of xterm.js.

**Comment by jianyunt** [comment]
After digging it further, I think the problem about the gabbled data likely come from PSReadline. See https://github.com/lzybkr/PSReadLine/issues/658

**Comment by jianyunt** [comment]
This fix gets fixed by corefx https://github.com/dotnet/corefx/issues/29669 and fix is in GA 6.1
